### ISSUE TITLE
REST - Setting `payment_method_title` on an order via PUT uses the wrong field.

### ISSUE THREAD

**Issue by larssn** [woocommerce/woocommerce#13796]
## EXPLANATION OF THE ISSUE

Setting `payment_method_title` on an order via PUT uses the wrong field, namely `payment_method`.

This is using the `wc/v1` endpoint

## STEPS TO REPRODUCE THE ISSUE

1. Create a simple order via REST. Leave `payment_method` and `payment_method_title` unset.
2. Do an PUT of the order, setting `payment_method` to the payment method's slug, and `payment_method_title` to something else.
3. The order returned in the response will have the same value for both `payment_method` and `payment_method_title`, namely the value from `payment_method`. So `payment_method_title` is basically discarded.

PUT (short version).

```
{
   "payment_method": "bacs",
   "payment_method_title": "Bank Transfer"
}
```

The returned order has:

```
{
   "payment_method": "bacs",
   "payment_method_title": "bacs"
}
```

## SYSTEM STATUS REPORT

```
### WordPress Environment ###

Home URL: Irrelevant
Site URL: Irrelevant
WC Version: 2.6.14
Log Directory Writable: ✔
WP Version: 4.7.3
WP Multisite: ✔
WP Memory Limit: 256 MB
WP Debug Mode: ✔
WP Cron: –
Language: da_DK

### Server Environment ###

Server Info: nginx/1.11.10
PHP Version: 7.0.17-2+deb.sury.org~xenial+1
PHP Post Max Size: 32 MB
PHP Time Limit: 240
PHP Max Input Vars: 1000
cURL Version: 7.47.0
OpenSSL/1.0.2k

SUHOSIN Installed: –
MySQL Version: 5.6.27
Max Upload Size: 1 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ❌ Your server does not have the SoapClient class enabled - some gateway plugins which use SOAP may not work as expected.
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 2.6.14
: 
woocommerce_sessions: ✔
woocommerce_api_keys: ✔
woocommerce_attribute_taxonomies: ✔
woocommerce_downloadable_product_permissions: ✔
woocommerce_order_items: ✔
woocommerce_order_itemmeta: ✔
woocommerce_tax_rates: ✔
woocommerce_tax_rate_locations: ✔
woocommerce_shipping_zones: ✔
woocommerce_shipping_zone_locations: ✔
woocommerce_shipping_zone_methods: ✔
woocommerce_payment_tokens: ✔
woocommerce_payment_tokenmeta: ✔
MaxMind GeoIP Database: ✔

### Active Plugins (6) ###

Carbon Fields: by htmlburger – 1.5
nuvoApp: by nuvoPoint – 1.16.0
nuvoBook: by nuvoPoint – 2.10.1
nuvoTakeAway: by nuvoPoint – 2.3.2
WooCommerce Bookings: by Automattic – 1.9.13
WooCommerce: by WooThemes – 2.6.14
WordPress MU Domain Mapping: by Donncha O Caoimh – 0.5.5.1
WP Pusher: by WP Pusher – 2.4.2
Yoast SEO: by Team Yoast – 4.5
S3 Uploads: by Human Made Limited – 2.0.0-beta2
nuvoSimplePackageInstaller_WPP: by nuvoPoint – 1.0.0
Nginx Helper: by rtCamp – 1.9.8
WP REST API - OAuth 2.0 Server: by justingreerbbi
ap.koponen – 1.0.1

WP Crontrol: by John Blackbourn & Edward Dale – 1.4
Jetpack by WordPress.com: by Automattic – 4.7.1

### Settings ###

Force SSL: –
Currency: DKK (DKK)
Currency Position: left_space
Thousand Separator: .
Decimal Separator: ,
Number of Decimals: 2

### API ###

API Enabled: ✔

### WC Pages ###

Shop Base: #3 - /butik/
Cart: #4 - /kurv/
Checkout: #5 - /kasse/
My Account: #6 - /min-konto/

### Taxonomies ###

Product Types: booking (booking)
external (external)
grouped (grouped)
simple (simple)
variable (variable)


### Theme ###

Name: Twenty Sixteen
Version: 1.3
Author URL: https://wordpress.org/
Child Theme: ❌ – If you're modifying WooCommerce on a parent theme you didn't build personally
then we recommend using a child theme. See: How to create a child theme

WooCommerce Support: ✔

### Templates ###

Overrides: –
```

**Comment by mikejolley** [comment]
They are different fields - payment_method_title does not change payment_method.

**Comment by larssn** [comment]
Thats not the issue.

**Comment by mikejolley** [comment]
If you mean that in reverse:

> The order returned in the response should have the same value for both payment_method and payment_method_title, namely the value from payment_method.

I can tell you updating a payment method does also set the title (pulls it from the gateway). Is that what you mean?

**Comment by larssn** [comment]
Well I must have written it in an unclear way then.

I mean i PUT this (short version).

```
{
   "payment_method": "bacs",
   "payment_method_title": "Bank Transfer"
}
```

The returned order has:

```
{
   "payment_method": "bacs",
   "payment_method_title": "bacs"
}
```

**Comment by mikejolley** [comment]
Confirmed this is broken in 2.6.x ( @claudiosanches FYI ), but is it resolved in 3.0 in both v1 and v2 apis.
